src/main/java/racinggame/model/ProgressRecord.java

```diff
@@ -0,0 +1,18 @@
+package racinggame.model;
+
+public record ProgressRecord(JoinCars joinCars) {
```

Comment:
JoinCars을 전달한 후 다시 JoinCars을 생성하기보다 다음과 같이 구현하면 어떨까?

JoinCars에 `ProgressRecord toProcessRecord()` 메서드를 추가하고 ProgressRecord는 JoinCars가 아니라 List<Car>를 생성자의 인자로 받으면 어떨까?


```suggestion
public record ProgressRecord(List<Car> cars) {
```

앞의 List<ProgressRecord>을 전달하는 부분도 비슷함.
상태 값을 가지는 객체에 메시지를 보내 해당 객체가 현재 상태 값을 기반으로 ProgressRecord 같은 객체를 생성하도록 사고의 전환을 해보면 어떨까?
이렇게 구현했을 때 테스트 코드는 어떻게 달라지는지 고민해 볼 것을 추천